src/lazy/encoder/value_writer.rs

```diff
@@ -38,46 +38,6 @@ pub trait AnnotatableValueWriter: Sized {
 
     /// Performs no operations and returns a [`ValueWriter`].
     fn without_annotations(self) -> Self::ValueWriter;
-
-    // Users can call `ValueWriter` methods on the `AnnotatedValueWriter` directly. Doing so
-    // will implicitly call `without_annotations`.
```

Comment:
🗺️ These default delegated methods mimicked those on `ValueWriter`, allowing users to call them on an `AnnotatableValueWriter` without first calling `with_annotations` or `without_annotations`. However, an `AnnotatableValueWriter` could not be passed to methods that required an implementation of `ValueWriter`.